src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -706,17 +709,20 @@ public void valueChanged(ListSelectionEvent evt) {
                         if (objects.length == 1
                             && ((File)objects[0]).isDirectory()
                             && chooser.isTraversable(((File)objects[0]))
-                            && (useSetDirectory || !fsv.isFileSystem(((File)objects[0])))) {
+                            && (useSetDirectory
+                                || (!fsv.isFileSystem(((File)objects[0]))
+                                    && !Files.isSymbolicLink(((File)objects[0]).toPath())))) {
                             setDirectorySelected(true);
                             setDirectory(((File)objects[0]));
                         } else {
                             ArrayList<File> fList = new ArrayList<File>(objects.length);
                             for (Object object : objects) {
-                                File f = (File) object;
+                                File f = (File)object;
                                 boolean isDir = f.isDirectory();
+                                Path path = f.toPath();
                                 if ((chooser.isFileSelectionEnabled() && !isDir)
                                     || (chooser.isDirectorySelectionEnabled()
-                                        && fsv.isFileSystem(f)
+                                        && (fsv.isFileSystem(f) || Files.isSymbolicLink(path))
```

Comment:
You can inline `f.toPath()` in line 725 and remove the local variable `path` like this:

```java
                                        && (fsv.isFileSystem(f) || Files.isSymbolicLink(f.toPath()))
```